src/Groups/MatGrp.jl

```diff
@@ -0,0 +1,790 @@
+import AbstractAlgebra: MatElem, matrix, MatSpace, parent_type, Ring, RingElem
+import Hecke: base_ring, det, fmpz, fq_nmod, FqNmodFiniteField, nrows, tr, trace
+import GAP: FFE
+
+export
+    general_linear_group,
+    mat_elem_type,
+    matrix_group,
+    MatrixGroup,
+    MatrixGroupElem,
+    omega_group,
+    orthogonal_group,
+    ring_elem_type,
+    special_linear_group,
+    special_orthogonal_group,
+    special_unitary_group,
+    symplectic_group,
+    unitary_group,
+    GL, GO, GU, SL, SO, Sp, SU
+
+
+
+abstract type AbstractMatrixGroupElem <: GAPGroupElem{GAPGroup} end
+
+# NOTE: always defined are deg, ring and at least one between { X, gens, descr }
+# NOTE: the fields ring_iso and mat_iso are always defined if the field X is
+"""
+    MatrixGroup{RE<:RingElem, T<:MatElem{RE}} <: GAPGroup
+Type of groups `G` of `n x n` matrices over the ring `R`, where `n = degree(G)` and `R = base_ring(G)`.
+
+At the moment, only rings of type `FqNmodFiniteField` are supported.
+"""
+mutable struct MatrixGroup{RE<:RingElem, T<:MatElem{RE}} <: GAPGroup
+   deg::Int
+   ring::Ring
+   X::GapObj
+   gens::Vector{<:AbstractMatrixGroupElem}
+   descr::Symbol                       # e.g. GL, SL, symbols for isometry groups
+   ring_iso::GenRingIso
```

Comment:
It is actually unclear to me why `ring_iso` is needed -- aren't you only every using `mat_iso`? I had a look at the code and my impression was that the only place using `ring_iso` is `assign_from_description`, and that only uses it to get `G.ring_iso.codomain`.

Ah, and also `@testset "Oscar-GAP relationship" begin` accesses it -- but I think that one could (and should) just be rewritten to test `mat_iso` instead (which is not tested there at all right now, as far as I could tell?); and I don't just mean that one should replace `G.ring_iso` by `G.mat_iso.fr`, but rather that the `mat_iso` function should be tested by mapping matrices back and forth, as that is the workhorse of the conversion, isn't it?

Reply:
Done it. I just removed the field `:ring_iso`.